.github/workflows/deploy.yaml

```diff
@@ -99,7 +99,10 @@ jobs:
           python-version-file: ".python-version"
           cache: "pip"
 
-      - run: pip install -r requirements/infra/requirements.txt
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
```

Comment:
I think this step should maybe come before `actions/setup-python@v4`, and `setup-python` should have `cache: "poetry"`.

Also, do we need this dependency? Couldn't this be `run: pip install poetry`?